docs/content/guides/dagster/managing-ml.mdx

```diff
@@ -0,0 +1,292 @@
+from dagster import MetadataValue
+import seaborn
+import matplotlib.pyplot as plt
+import base64
+from io import BytesIO
+
+
+def make_plot(eval_metric):
+    plt.clf()
+    training_plot = seaborn.lineplot(eval_metric)
+    fig = training_plot.get_figure()
+    buffer = BytesIO()
+    fig.savefig(buffer)
+    image_data = base64.b64encode(buffer.getvalue())
+    return MetadataValue.md(f"![img](data:image/png;base64,{image_data.decode()})")
+```
+
+In this example, a dictionary is used called `metadata` to store the Markdown plots and the score value in Dagster.
+
+```python file=/guides/dagster/managing_ml/managing_ml_code.py startafter=metadata_use_start endbefore=metadata_use_end
+from dagster import asset
+import xgboost as xgb
+from sklearn.metrics import mean_absolute_error
+
+
+@asset
+def xgboost_comments_model(transformed_training_data, transformed_test_data):
+    transformed_X_train, transformed_y_train = transformed_training_data
+    transformed_X_test, transformed_y_test = transformed_test_data
+    # Train XGBoost model, which is a highly efficient and flexible model
+    xgb_r = xgb.XGBRegressor(
+        objective="reg:squarederror", eval_metric=mean_absolute_error, n_estimators=20
+    )
+    xgb_r.fit(
+        transformed_X_train,
+        transformed_y_train,
+        eval_set=[(transformed_X_test, transformed_y_test)],
+    )
+
+    ## plot the mean absolute error values as the training progressed
+    metadata = {}
+    for eval_metric in xgb_r.evals_result()["validation_0"].keys():
+        metadata[f"{eval_metric} plot"] = make_plot(
+            xgb_r.evals_result_["validation_0"][eval_metric]
+        )
+    # keep track of the score
+    metadata["score (mean_absolute_error)"] = xgb_r.evals_result_["validation_0"][
+        "mean_absolute_error"
+    ][-1]
+
+    return Output(xgb_r, metadata=metadata)
+```
+
+In the Dagster UI, the `xgboost_comments_model` has the metadata rendered. Numerical values, such as the `score (mean_absolute_error)` will be logged and plotted for each materialization, which can be useful to understand the score over time for machine learning models.
+
+<Image
+alt="alt"
+src="/images/guides/managing_ml/managing_ml_ui.png"
+width={1188}
+height={541}
+/>
+
+The Markdown plots are also available to inspect the evaluation metrics during the training cycle by clicking on **\[Show Markdown]**:
+
+<Image
+alt="alt"
+src="/images/guides/managing_ml/plot_ui.png"
+width={1188}
+height={541}
+/>
+
+---
+
+## Tracking model history
+
+Viewing previous versions of a machine learning model can be useful to understand the evaluation history or referencing a model that was used for inference. Using Dagster will enable you to understand - what the data was that was used to train the model, when it was refreshed, what code version and finally what version of the ML model was used to generate whatever predictions it is used for the predicted values.
```

Comment:
```suggestion
Viewing previous versions of a machine learning model can be useful to understand the evaluation history or referencing a model that was used for inference. Using Dagster will enable you to understand:

- What data was used to train the model
- When the model was refreshed
- The code version and ML model version was used to generate the predictions used for predicted values
```

This is really good info, but I think putting it into list format makes it easier to read and absorb. I think the last item here is worded a bit awkwardly though, but I wasn't entirely sure what it was trying to say originally.